dts/bindings/sensor/st,stm32-qdec.yaml

```diff
@@ -0,0 +1,45 @@
+# Copyright (c) 2022, Valerio Setti <vsetti@baylibre.com
+# SPDX-License-Identifier: Apache-2.0
+
+description: STM32 quadrature decoder
+
+compatible: "st,stm32-qdec"
+
+include:
+    - name: base.yaml
+    - name: pinctrl-device.yaml
+
+properties:
+     pinctrl-0:
+       required: true
+
+     pinctrl-names:
+       required: true
+
+     input-polarity-inverted:
+       type: boolean
+       required: false
+       description: encoder is triggered by a falling edge on the input pin
+
+     input-filter-level:
+       type: int
+       required: true
+       description: intensity of filtering on the input
+       default: 0
+       enum:
+         - 0  # No filter
+         - 1  # FDIV1_N2
+         - 2  # FDIV1_N4
+         - 3  # FDIV1_N8
+         - 4  # FDIV2_N6
+         - 5  # FDIV2_N8
+         - 6  # FDIV4_N6
+         - 7  # FDIV4_N8
+         - 8  # FDIV8_N6
+         - 9  # FDIV8_N8
+         - 10 # FDIV16_N5
+         - 11 # FDIV16_N6
+         - 12 # FDIV16_N8
+         - 13 # FDIV32_N5
+         - 14 # FDIV32_N6
+         - 15 # FDIV32_N8
```

Comment:
This could use a better description. Also, please note that comments are not rendered in zephyr docs.
See https://docs.zephyrproject.org/latest/build/dts/api/bindings/counter/st%2Cstm32-counter.html#dtbinding-st-stm32-counter.